Gram CLI v0.1.0

cjea opened the pull request:
## Description
Beta release of `gram` CLI. See the README for release instructions.

## Details
_TODO: fill out details for the usage, examples, and architecture._

simplesagar commented on server/cmd/cli/gram/README.md:
```diff
@@ -0,0 +1,49 @@
+$ go run . --help
+
+NAME:
+   gram - Remote MCP management
```
```suggestion
   gram - CLI to interact with the [Gram Platform](..._link to cli docs_....) 
```

cjea commented:
I exercised the release flow locally in case anyone would like to try it for review.
```bash
$ mise run cli:release
[cli:release] $ ~/code/speakeasy/gram/.mise-tasks/cli/release
[cli:tidy] $ ~/code/speakeasy/gram/.mise-tasks/cli/tidy
Updating CLI go.mod dependency.
Updated upstream dependency: github.com/speakeasy-api/gram/server@v0.0.0-20250916221109-1ba512972c74
Updated /Users/babe/code/speakeasy/gram/server/cmd/cli/gram/go.mod
[cli:version] $ ~/code/speakeasy/gram/.mise-tasks/cli/version

Current version: v0.0.1-alpha (/code/speakeasy/gram/server/cmd/cli/gram/version/version.go)
Is v0.0.1-alpha the correct version to tag? Have you bumped it if needed? (y/N): y

+ git tag server/cmd/cli/gram/v0.0.1-alpha
+ set +x
Successfully tagged commit with server/cmd/cli/gram/v0.0.1-alpha
You can now push the tag with: git push origin server/cmd/cli/gram/v0.0.1-alpha

$ git push origin server/cmd/cli/gram/v0.0.1-alpha
```

```
$ go install github.com/speakeasy-api/gram/server/cmd/cli/gram@v0.0.1-alpha
go: downloading github.com/speakeasy-api/gram/server/cmd/cli/gram v0.0.1-alpha
$ gram -v
gram version v0.0.1-alpha
```

cjea commented on server/cmd/cli/gram/env/env.go:
```diff
@@ -0,0 +1,81 @@
+		panic(fmt.Errorf("missing env: %s", key))
+	}
+
+	return val
```
This is handled gracefully when starting with [env.APIKeyMissing](https://github.com/speakeasy-api/gram/pull/262/files#diff-595900f92db186357fc632deb1a824aa686dce61630176cbc4ca21fcd9e14a4bR93). Later in the program, this value should never be empty.

```go
if env.APIKeyMissing() {
  return fmt.Errorf(
	"API key not set. Please set the %s environment variable and retry”,
	env.VarNameProducerKey,
  )
}
```

ryan-timothy-albert reviewed (commented):
Personally I like apply over push. Since we are sort of going for the declarative terraformish model.

ryan-timothy-albert reviewed (approved):
I think we are good to merge and continue to iterate rather than needing to wait.

As long as we stay below a v1.0.0 version and don't yet fully document it as released we can comfortably make breaking changes to this while we still iron out a few things.

cjea commented:
> I think we are good to merge and continue to iterate rather than needing to wait.

Bombs away!
